controllers/dscinitialization/utils.go

```diff
@@ -18,132 +19,184 @@ import (
-			labelPatch := `{"metadata":{"labels":{"openshift.io/cluster-monitoring":"true", "pod-security.kubernetes.io/enforce":"baseline","opendatahub.io/generated-namespace": "true"}}}`
 
-			err = r.Patch(ctx, foundMonitoringNamespace, client.RawPatch(types.MergePatchType, []byte(labelPatch)))
-			if err != nil {
-				return err
-			}
+	// managed cluster need to patch its namespaces
+	if platform == cluster.ManagedRhoai {
+		// Patch default app namespace for Managed cluster
+		err := r.patchAppNS(ctx, nsName, log)
+		if err != nil {
+			log.Error(err, "error patch application namespace for managed cluster")
+			return err
 		}
 	}
+	// Patch monitoring namespace
+	err := r.patchMonitoringNS(ctx, dscInit, log)
+	if err != nil {
+		log.Error(err, "error patch monitoring namespace for managed cluster")
+		return err
+	}
 
 	// Create default NetworkPolicy for the namespace
-	err = r.reconcileDefaultNetworkPolicy(ctx, name, dscInit, platform)
+	err = r.reconcileDefaultNetworkPolicy(ctx, nsName, dscInit, platform, log)
 	if err != nil {
-		log.Error(err, "error reconciling network policy ", "name", name)
+		log.Error(err, "error reconciling network policy ", "name", nsName)
 		return err
 	}
 
 	// Create odh-common-config Configmap for the Namespace
-	err = r.createOdhCommonConfigMap(ctx, name, dscInit)
+	err = r.createOdhCommonConfigMap(ctx, nsName, dscInit, log)
 	if err != nil {
 		log.Error(err, "error creating configmap", "name", "odh-common-config")
 		return err
 	}
 
 	// Create default Rolebinding for the namespace
-	err = r.createDefaultRoleBinding(ctx, name, dscInit)
+	err = r.createDefaultRoleBinding(ctx, nsName, dscInit, log)
 	if err != nil {
-		log.Error(err, "error creating rolebinding", "name", name)
+		log.Error(err, "error creating rolebinding", "name", nsName)
 		return err
 	}
 	return nil
 }
 
-func (r *DSCInitializationReconciler) createDefaultRoleBinding(ctx context.Context, name string, dscInit *dsciv1.DSCInitialization) error {
-	log := logf.FromContext(ctx)
+func (r *DSCInitializationReconciler) createAppNamespace(ctx context.Context, nsName string) error {
+	desiredDefaultNS := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{Name: nsName},
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, desiredDefaultNS, func() error {
+		resources.SetLabel(desiredDefaultNS, labels.ODH.OwnedNamespace, labels.True) // this indicate when uninstall, namespace will be deleted
+		resources.SetLabel(desiredDefaultNS, labels.SecurityEnforce, "baseline")
+		return nil
+	})
+	return err
+}
+
+func (r *DSCInitializationReconciler) validateCustomizeAppNS(ctx context.Context, nsName string, log logr.Logger) error {
+	// user defined application namespace
+	// Check if application namespace has label "opendatahub.io/application-namespace:true"
+	// we do not remove generated-namespace label if it exists, it is up to user to decide if they want namespace to be deleted or not during uninstallation.
```

Comment:
I don't think this comment is correct, as it does not perform any validation.

Reply:
but it does a Get() call with label, so if it does not return object, then it fails "check", right?

Reply:
my understanding is that the label set in the object does not matter at all for a get operation, the look up is performed using the object key